ssh-server/main.go

```diff
@@ -113,38 +118,96 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	}
 	defer channel.Close()
 
+	allow := map[string]bool{"shell": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	defer close(resize)
+	done := make(chan error)
+	for {
+		select {
+		case err := <-done:
+			if err != nil {
+				return fmt.Errorf("running pty: %v", err)
+			}
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				r := bytes.NewReader(req.Payload)
+				term, err := readString(r)
+				if err != nil {
+					return fmt.Errorf("parsing TERM environment variable value: %v", err)
+				}
+
+				if err := readWindowSize(r, resize); err != nil {
+					return fmt.Errorf("reading window size: %v", err)
+				}
+
+				go func() {
+					done <- runPTY(channel, term, resize)
+				}()
+			case "window-change":
+				if err := readWindowSize(bytes.NewReader(req.Payload), resize); err != nil {
+					return fmt.Errorf("reading window size: %v", err)
+					continue
```

Comment:
continue after return doesn't mean anything... out of curiousity, does go vet catch this? (i think it might)